node/src/components/block_synchronizer.rs

```diff
@@ -1252,15 +1252,29 @@ impl BlockSynchronizer {
             }
         }
 
-        BlockSynchronizerProgress::Syncing(
-            builder.block_hash(),
-            builder.block_height(),
-            builder.last_progress_time().max(
-                self.global_sync
-                    .last_progress()
-                    .unwrap_or_else(Timestamp::zero),
-            ),
-        )
+        let last_progress_time = builder.last_progress_time().max(
+            self.global_sync
+                .last_progress()
+                .unwrap_or_else(Timestamp::zero),
+        );
+
+        if last_progress_time.elapsed() > self.config.progress_stall_threshold_interval {
+            BlockSynchronizerProgress::Stalled(
+                builder.block_hash(),
+                builder.block_height(),
+                last_progress_time,
+            )
+        } else {
+            BlockSynchronizerProgress::Syncing(
+                builder.block_hash(),
+                builder.block_height(),
+                builder.last_progress_time().max(
+                    self.global_sync
+                        .last_progress()
+                        .unwrap_or_else(Timestamp::zero),
```

Comment:
Could have used `last_progress_time` here too.

Reply:
Addressed in 961750e61cca7014966adcf3785324191038f2f7